tests/switch.t

```diff
@@ -0,0 +1,144 @@
+local test = require 'test'
+
+function dotest(fn, a, b, c, d)
+  test.eq(fn(1), a)
+  test.eq(fn(2), b)
+  test.eq(fn(3), c)
+  test.eq(fn(4), d)
+end
+
+terra foo(a: int): int
+  switch a do
+    case 1 then
+      return 2
+    case 2 then
+      return 5
+    end
+  else
+    return 3
+  end
+end
+
+dotest(foo, 2, 5, 3, 3)
+
+terra empty(a: int): int
+  switch a do
+  end
+  return -1
+end
+
+dotest(empty, -1, -1, -1, -1)
+
+terra single(a: int): int
+  switch a do
+    case 1 then 
+      return 2
+    end
+  end
+  return -1
+end
+
+dotest(single, 2, -1, -1, -1)
+
+terra justelse(a: int): int
+  switch a do
+    else
+      return 3
```

Comment:
Following on the above, should we require `else` to have its own `end`? (For consistency with when we metaprogram the `else`.)

Reply:
The `end` here is intended to conclude the entire case/case/else sequence, it's not just for cases. Once we allow `else` to be metaprogrammed, it would need an `end` statement when inside a quote, but there's no reason to give it one inside the switch statement.

Reply:
Else is already metaprogrammable by just putting a splice inside the `else` block.